sklearn/metrics/ranking.py

```diff
@@ -255,10 +256,24 @@ def _binary_roc_auc_score(y_true, y_score, sample_weight=None):
                                         sample_weight=sample_weight)
         return auc(fpr, tpr, reorder=True)
 
-    return _average_binary_score(
-        _binary_roc_auc_score, y_true, y_score, average,
-        sample_weight=sample_weight)
-
+    if type_of_target(y_true) != "multiclass":
+        return _average_binary_score(
+            _binary_roc_auc_score, y_true, y_score, average,
+            sample_weight=sample_weight)
+    else:
```

Comment:
Why can't you use ``MultiLabelEncoder`` for the OVR case?
You could change the above to ``multiclass != "ovo"`` and add a call to ``MultiLabelEncoder`` for the multi-class case.